tests/MongoDB.Driver.Core.Tests/Core/Compression/ZstandardNativeTests.cs

```diff
@@ -0,0 +1,172 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core.Compression.Zstandard;
+using MongoDB.Driver.Core.Misc;
+using Xunit;
+
+namespace MongoDB.Driver.Core.Tests.Core.Compression
+{
+    public class ZstandardNativeTests
+    {
+        #region static
+        // private constants
+        private const string __testMessagePortion = @"Two households, both alike in dignity,
+        In fair Verona, where we lay our scene,
+        From ancient grudge break to new mutiny,
+        Where civil blood makes civil hands unclean.
+            From forth the fatal loins of these two foes
+            A pair of star-cross'd lovers take their life;
+        Whose misadventured piteous overthrows
+        Do with their death bury their parents' strife.
+            The fearful passage of their death-mark'd love,
+        And the continuance of their parents' rage,
+        Which, but their children's end, nought could remove,
+        Is now the two hours' traffic of our stage;
+        The which if you with patient ears attend,
+        What here shall miss, our toil shall strive to mend.";
+
+        // private static fields
+        private static readonly byte[] __bigMessage = GenerateBigMessage(135000); // bigger than recommended size for one native operation
+
+        // private static methods
+        private static byte[] GenerateBigMessage(int size)
+        {
+            var resultBytes = new List<byte>();
+            var messagePortionBytes = Encoding.ASCII.GetBytes(__testMessagePortion);
+            while (resultBytes.Count < size)
+            {
+                resultBytes.AddRange(messagePortionBytes);
+            }
+            return resultBytes.ToArray();
+        }
+        #endregion
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Compressor_should_decompress_the_previously_compressed_message([Range(1, 22)] int compressionLevel)
+        {
+            var messageBytes = __bigMessage;
+
+            var compressedBytes = Compress(messageBytes, compressionLevel);
+            compressedBytes.Length.Should().BeLessThan(messageBytes.Length / 2);
+
+            var decompressedBytes = Decompress(compressedBytes);
+            decompressedBytes.ShouldBeEquivalentTo(messageBytes);
```

Comment:
I'm not sure `ShouldBeEquivalent` is correct. Normally comparisons that have "Equivalent" in the name ignore order.

In other places we have used:

    decompressedBytes.SequenceEqual(messageBytes).Should().BeTrue();

Reply:
As I remember, it works correctly but I will double check in the next commit

Reply:
It may work correctly, but since "EquivalentTo" usually implies order should be ignored it might be worth changing.

Reply:
The usual way to compare two arrays in FluentAssertions is:

    decompressedBytes.Should().Equal(messageBytes);

Note: that's `Equal` not `Equals`.